schematic/store/synapse.py

```diff
@@ -788,18 +799,47 @@ def fill_in_entity_id_filename(
         # the columns Filename and entityId are assumed to be present in manifest schema
         # TODO: use idiomatic panda syntax
         if dataset_files:
+            all_files = self._get_file_entityIds(
+                dataset_files=dataset_files, only_new_files=False, manifest=manifest
+            )
             new_files = self._get_file_entityIds(
                 dataset_files=dataset_files, only_new_files=True, manifest=manifest
             )
 
-            # update manifest so that it contains new dataset files
+            all_files = pd.DataFrame(all_files)
             new_files = pd.DataFrame(new_files)
+
+            # update manifest so that it contains new dataset files
             manifest = (
                 pd.concat([manifest, new_files], sort=False)
                 .reset_index()
                 .drop("index", axis=1)
             )
 
+            # Reindex manifest and new files dataframes according to entityIds to align file paths and metadata
+            manifest_reindex = manifest.set_index("entityId")
+            all_files_reindex = all_files.set_index("entityId")
+            all_files_reindex_like_manifest = all_files_reindex.reindex_like(
+                manifest_reindex
+            )
+
+            # Check if individual file paths in manifest and from synapse match
+            file_paths_match = (
+                manifest_reindex["Filename"]
+                == all_files_reindex_like_manifest["Filename"]
+            )
+
+            # If all the paths do not match, update the manifest with the filepaths from synapse
+            if not file_paths_match.all():
+                manifest_reindex.loc[
+                    ~file_paths_match, "Filename"
+                ] = all_files_reindex_like_manifest.loc[~file_paths_match, "Filename"]
+
+                # reformat manifest for further use
+                manifest = manifest_reindex.reset_index()
+                entityIdCol = manifest.pop("entityId")
+                manifest.insert(len(manifest.columns), "entityId", entityIdCol)
```

Comment:
I'm coming at this from a naive angle, but it seems what is on synapse is the source of truth. Instead of spending the compute and doing a comparison, why not just do a join and replace with what's on Synapse?  This would probably simplify the code, but I'm most likely missing something.

Reply:
@thomasyu888 this is me being conservative from a user experience standpoint. The files that are returned from synapse are sorted in alphabetical order. What I'm thinking of is the case where a user has already uploaded a manifest and annotated files, and later more files are uploaded that may not alphabetically come after the existing files on synapse. 

In this case, if we just take and copy what is on synapse, the rows for the new files that need to be annotated would be scattered separately amongst the already annotated files. What I'm trying to cover by the reindexing, matching, and the resetting is users being confused as to where the rows are for the new files they have to annotate. Matching based on entityId ensures that the files are in the same order so we can do a safer filename swap (as opposed to the case where files are in a different order in the manifest and on synapse) and then the reindexing preserves the old behavior where new files always come after the already annotated files that exists on synapse.

Reply:
Makes sense! Thanks for the thoughtful explanation!

Reply:
@thomasyu888 yeah no problem!